components/UploadForm.vue

```diff
@@ -0,0 +1,671 @@
+  try {
+    uploadStatus.value = 'loading'
+    const results = []
+    for (const record of fileRecords.value) {
+      await new Promise(resolve => setTimeout(resolve, 100))
+
+      const priceResult = await estimateBundlrFilePrice({
+        fileSize: record.fileBlob?.size || 0,
+        ipfsHash: record.ipfsHash
+      })
+      results.push({
+        ...priceResult,
+        ipfsHash: record.ipfsHash
+      })
+    }
+
+    let totalFee = new BigNumber(0)
+    results.forEach((result) => {
+      const { address, arweaveId, LIKE, ipfsHash } = result
+      if (LIKE) {
+        totalFee = totalFee.plus(new BigNumber(LIKE))
+        arweaveFeeMap.value[ipfsHash] = LIKE
+      }
+      if (arweaveId) {
+        sentArweaveTransactionInfo.value.set(ipfsHash, {
+          transactionHash: '',
+          arweaveId
+        })
+        const metadata = epubMetadataList.value.find(
+          (data: any) => data.thumbnailIpfsHash === ipfsHash
+        )
+        if (metadata) {
+          metadata.thumbnailArweaveId = arweaveId
+        }
+      }
+      if (!arweaveFeeTargetAddress.value) {
+        arweaveFeeTargetAddress.value = address
+      }
+    })
+
+    arweaveFee.value = totalFee
+  } catch (err) {
+    console.error(err)
+  } finally {
+    uploadStatus.value = ''
+  }
+}
+
+const submitToArweave = async (record: any): Promise<void> => {
+  const existingData =
+    sentArweaveTransactionInfo.value.get(record.ipfsHash) || {}
+  const { transactionHash, arweaveId: uploadArweaveId } = existingData
+  if (uploadArweaveId) {
+    return
+  }
+
+  if (!record.fileBlob) {
+    return
+  }
+  // open sign dialog
+  let txHash = transactionHash
+  if (!txHash) {
+    txHash = await sendArweaveFeeTx(record)
+    if (!txHash) {
+      throw new Error('TRANSACTION_NOT_SENT')
+    }
+  }
+
+  try {
+    const arrayBuffer = await record.fileBlob.arrayBuffer()
+    const buffer = Buffer.from(arrayBuffer)
+    const { arweaveId, arweaveLink } = await uploadSingleFileToBundlr(buffer, {
+      fileSize: record.fileBlob?.size || 0,
+      ipfsHash: record.ipfsHash,
+      fileType: record.fileType as string,
+      txHash,
+      token: token.value
+    })
+
+    if (arweaveId) {
+      const uploadedData =
+        sentArweaveTransactionInfo.value.get(record.ipfsHash) || {}
+      sentArweaveTransactionInfo.value.set(record.ipfsHash, {
+        ...uploadedData,
+        arweaveId,
+        arweaveLink
+      })
+      if (record.fileName.includes('cover.jpeg')) {
+        const metadata = epubMetadataList.value.find(
+          (file: any) => file.thumbnailIpfsHash === record.ipfsHash
+        )
+        if (metadata) {
+          metadata.thumbnailArweaveId = arweaveId
+        }
+      }
+      emit('arweaveUploaded', { arweaveId, arweaveLink })
+      // close sign dialog
+    } else {
+      isOpenWarningSnackbar.value = true
+      error.value = 'IscnRegisterForm.error.arweave'
+      throw new Error(error.value)
+    }
```

Comment:
Prefer throw early for error guard like this, ie

```
if (!arweaveId) {
  ...
  throw new Error()
}
// continue normal logic
```

Reply:
[🥅 Early return if missing arweaveId](https://github.com/likecoin/nft-book-press/pull/290/commits/17aab909cde552b3c4971ccfc5be740ecdf7dde0)